wikitexthtml/render/preprocess.py

```diff
@@ -73,6 +82,62 @@ def hr_line(instance: WikiTextHtml, body: str) -> str:
     return regex.sub(r"\n----+\n", r"\n\n<hr />\n", body)
 
 
+def highlight_syntax(instance: WikiTextHtml, body: str) -> str:
+    # Use Pygments to generate highlighted code blocks.
+
+    # The SYNTAXHIGHLIGHT_PATTERN Generates 6 groups:
+    #   The first group (match[0]) is the entire match.
+    #   The third group ('lang') is the language.
+    #   The fourth group ('line') is a boolean to specify line numbering.
+    #   The fifth group ('linestart') is the line number to start at.
+    #   The sixth group ('code') is the code to highlight.
+
+    for match in reversed(list(SYNTAXHIGHLIGHT_PATTERN.finditer(body))):
+        lang = match.group("lang")
+        line_boolean = True if match.group("line") else False
+        line_start = int(match.group("lineStart")) if match.group("lineStart") else 1
+        content = match.group("code")
+
+        try:
+            # Try to find a lexer based on the text captured by the 'lang' group.
+            lexer = get_lexer_by_name(lang)
+        except ClassNotFound:
+            # Either there was no language specified, or the language is not supported by Pygments.
+            try:
+                lexer = guess_lexer(content)
+                if lang:
+                    instance.add_error(
+                        f"Syntax highlighting: unknown language '{lang}'. (Using best guess: {lexer.name}"
```

Comment:
My OCD triggered: missing ) at end of string :D Sorry ......

Reply:
whoops!